rfcs/rfc-0000-discriminators.md

```diff
@@ -0,0 +1,123 @@
+
+- Allow users who do not want to be easily discoverable to stay hidden.
+
+  Just a 4-digit discriminator provides 10,000 unique users.
+  Expanding on this, we could for example, let users hide discriminators in servers and similar privacy features.
+
+# Guide-level explanation
+
+Revolt would be switching to a new username system made up of three parts:
+
+```
+Hello, World!#123456
+^ your chosen name
+             ^ separator between chosen name and discriminator
+              ^ discriminator
+```
+
+Discriminators for new users are generated randomly on sign up while existing users will
+get a randomly generated discriminators upon the server software being upgraded.
+
+> Revolt treats usernames as case-insensitive so a full username such as "Jason#1234" is
+> effectively the same as "jasON#1234" but only "Jason#1234" will display in the UI.
+
+Users may:
+
+- Change the case of their username at any moment, such as from "Jason" to "jasON".
+
+  This will never affect their discriminator.
+
+- Change their username, but may have their discriminator regenerated if there is a conflict.
+
+  Such as if you are changing from "Jason#1234" to "Phil", but "Phil#1234" is taken.
+
+### User Opinion
+
+Prior discussions on the forums primarily seem to lean in favour of discriminators being
+implemented into Revolt, the internal team has not raised any complaints against it, and
+the primary community on Revolt ("Revolt Lounge") appears to be somewhat split on the
+issue.
+
+As a recent case study, Discord's change to unique usernames only has generated a lot of
+backlash and criticism from the community, it appears to be overwhemingly negative but
+these experiences are mostly coming from long-term users of Discord. The true opinion of
+the average user has not been gauged yet.
+
+### Usability
+
+This may make it more difficult to understand how to add users on Revolt, however this
+shouldn't be an issue as long as it's clearly communicated through the user interface.
+Revolt already allows a wide range of Unicode in usernames so we are definitely not
+making a significant UX hit towards the usability of the friend system.
+
+> There are further considerations listed at the end of this document which must be
+> addressed if this RFC were to be implemented.
+
+# Reference-level explanation
+
+WIP
+
+This is the technical section of the RFC, it should go over in detail:
+
+- Its interaction with other features
+- How this will be implemented
+- Corner or edge cases
+
+This section should reference the examples in the previous section and disect them in more detail.
+
+# Drawbacks
+
+WIP
+
+Why should this not be added.
+
+# Rationale and alternatives
+
+WIP
+
+- Why is this design the best
+- Are there alternative ways to solve this
+- Could this be done with existing features or existing solutions
+
+# Prior art
+
+This has been implemented before on other platforms:
+
+- Discord's discriminator system
```

Comment:
```suggestion
```
Discord is getting rid of discriminators

Reply:
It's still an implementation of this system so it's still good to keep here.

Reply:
if anything, it should just say "Discord's (soon to be defunct) discriminator system"